File: backend/models/deals.py
from pydantic import BaseModel
from typing import List, Optional


class Deal(BaseModel):
    #Model representing a single deal
    id: int
    title: str
    created_at: str
    firm: Optional[str] = ""
    asset_class: Optional[str] = "General"
    deal_status: Optional[str] = "Unknown"
    currency: Optional[str] = "USD"
    user_id: Optional[int] = 0
    deal_capital_seeker_email: Optional[str] = ""


class FileInfo(BaseModel):
    #model representing file information
    id: int | str
    name: str
    size: int
    url: str
    type: Optional[str] = ""
    download_url: str
    created_at: Optional[str] = ""
    state: Optional[str] = ""


class DealsRequest(BaseModel):
    #request model for fetching deals
    website: str
    token: str


class DealsResponse(BaseModel):
    #response model for deals list
    deals: List[Deal]
    total: int


class FilesRequest(BaseModel):
    """Request model for fetching deal files"""
    website: str
    token: str
    deal_id: int


class FilesResponse(BaseModel):
    """Response model for deal files"""
    files: List[FileInfo]
    total: int
    error: Optional[str] = None 

File: frontend/frontend/src/services/deals.ts
// Deals service and types
import { api } from "./api";

// Deal types
interface Deal {
  id: number;
  title: string;
  created_at: string;
  firm: string;
  asset_class: string;
  deal_status: string;
  currency: string;
  user_id: number;
  deal_capital_seeker_email: string;
}

interface FileInfo {
  id: number | string;
  name: string;
  size: number;
  url: string;
  type: string;
  download_url: string;
  created_at?: string;
  state?: string;
}

interface DealsResponse {
  deals: Deal[];
  total: number;
}

interface FilesResponse {
  files: FileInfo[];
  total: number;
  error?: string;
}

// Deals functions
export async function getDeals(): Promise<DealsResponse> {
  // Token and website are automatically added by interceptor
  const response = await api.post("/deals/list", {});
  return response.data;
}

export async function getDealFiles(dealId: number): Promise<FilesResponse> {
  const response = await api.post(`/deals/${dealId}/files`, {
    deal_id: dealId
  });
  return response.data;
} 

export type { Deal, FileInfo, DealsResponse, FilesResponse };